kornia/contrib/connected_components.py

```diff
@@ -51,8 +52,18 @@ def connected_components(image: Tensor, num_iterations: int = 100) -> Tensor:
     out = torch.arange(B * H * W, device=image.device, dtype=image.dtype).view((-1, 1, H, W))
     out[~mask] = 0
 
-    for _ in range(num_iterations):
-        out = F.max_pool2d(out, kernel_size=3, stride=1, padding=1)
-        out = torch.mul(out, mask)  # mask using element-wise multiplication
+    i = 0
+    while True:
+        i += 1
+        out_ = F.max_pool2d(out, kernel_size=3, stride=1, padding=1)
+        # mask using element-wise multiplication
+        out_ = torch.mul(out_, mask)
+        # stop if converged
+        if torch.all(out == out_):
+            break
+        out = out_
+        # reached if reached max iterations
+        if num_iterations > 0 and i == num_iterations:
```

Comment:
```suggestion
        if i >= num_iterations or error >= threshold:
```
i believe something like that should work. However, if we introduce this condition i'm afraid that we cannot support batch as it is. So, we might have to introduce a small logic that includes `vmap` @johnnv1 ?

Reply:
When using batch size greater than 1 the convergence should be reached when every element in the batch reaches it.